src/simsopt/mhd/spec.py

```diff
@@ -192,6 +722,12 @@ def set_dofs(self, x):
         self.inputlist.phiedge = x[0]
         self.inputlist.curtor = x[1]
 
+        # Update phiedge in profiles if necessary. This is important when taking
+        # profiles derivatives.
+        for ii, p in enumerate(self.profiles):
+            if not p is None:
+                self.profiles[ii].phiedge = x[0]
```

Comment:
See the issue around line 272 - I think `self.profiles` becomes out of sync if any profiles are set to new profiles. I think the following change here would resolve the issue:
~~~~py
        profiles = [
            self._volume_current_profile,
            self._interface_current_profile,
            self._pressure_profile,
            self._iota_profile,
            self._oita_profile,
            self._mu_profile,
            self._pflux_profile,
            self._tflux_profile,
            self._helicity_profile
	]
        for p in profiles:
            if p is not None:
                p.phiedge = x[0]
~~~~